apps/api/src/domain/models/book/entities/book.entity.ts

```diff
@@ -40,4 +38,28 @@ export class Book extends Resource {
         // TODO remove all casts like this
         this.pages = (pageDTOs as BookPage[]).map((pageDTO) => new BookPage(pageDTO));
     }
+
+    /**
+     *
+     * This relies on convention. It's a clever way to avoid a switch statement.
+     * TODO capitalize first letter
+     */
+    protected ispageRangeContextValid(context: PageRangeContext): Valid | InternalError {
+        if (isNullOrUndefined(context))
```

Comment:
Also, use proper camel case for the validator method names. And I wouldn't call them `is{x}valid` since I refactored to return something other than a boolean.

Reply:
done